Add documentation for production admiral deployment (#147)
istio-ecosystem/admiral: docs/Examples.md

## Example Installations & Demos

### Production Deployment

![](Admiral_Diagram.png)

> In order to run admiral in production environment, we will have two types of clusters:

Reviewer: ```suggestion
An admiral production set up would have two types of clusters:
```

> - one main cluster where admiral lives

Reviewer: ```suggestion
-  cluster where admiral runs called the `main` cluster
```

> - other remote clusters where admiral watches and monitors.

Reviewer: ```suggestion
- clusters which admiral watches, monitors and creates Istio CRs called the `remote` clusters
```

The requirements are different for the two types:

> - admiral namespace will exist in main cluster

Reviewer: ```suggestion
- admiral namespace will exist in the main cluster
```

- admiral-sync namespace will exist in remote clusters that admiral watches and monitors.

> 1\. Set necessary environment variables  

Reviewer: ```suggestion
The following steps show how you can install admiral in the main cluster and then provision secrets for it to watch the remote clusters. 

```While the process shown below using shell scripts is manual, automation can be built for i) installing admiral and its necessary resources (using a CD pipeline) ii) Provisioning secret for every new cluster created or upgraded as a post cluster create step```
 
1\. Set the necessary environment variables  
```

```bash
# Set main cluster env variable
export MAIN_CLUSTER=<path_to_kubeconfig_for_main_cluster>

# Set remote cluster env variable
export REMOTE_CLUSTER=<path_to_kubeconfig_for_remote_cluster>
```

2\. Install/Run Admiral in the main cluster

```bash
#Download and extract admiral

wget https://github.com/istio-ecosystem/admiral/releases/download/v1.0/admiral-install-v1.0.tar.gz
tar xvf admiral-install-v1.0.tar.gz

export ADMIRAL_HOME=./admiral-install-v1.0
```

```bash
#Install admiral
export KUBECONFIG=$MAIN_CLUSTER
$ADMIRAL_HOME/scripts/install_admiral.sh $ADMIRAL_HOME

```

3\. Add main cluster to Admiral's watcher

> Since there are most likely other contents living in the same cluster where admiral lives, admiral need to watch the cluster it's currently living in as well. This step can be skipped if Admiral has a dedicated cluster for Admiral only.

Reviewer: ```suggestion
Since there are most likely other workloads running in the same cluster where admiral lives, admiral needs to watch the cluster it's currently living in as well. This step can be skipped if Admiral runs in a dedicated cluster.
```

> Let admiral monitor the cluster it lives in by exchanging secret with itself.

Reviewer: ```suggestion
Let admiral monitor the cluster it lives in by using the secret to talk to the API server of the cluster where it runs.
```
